noir-projects/noir-protocol-circuits/crates/types/src/merkle_tree/indexed_tree.nr

```diff
@@ -26,6 +26,8 @@ pub fn batch_insert<Value, Leaf, SubtreeWidth, SiblingPathLength, SubtreeHeight,
     // A permutation to the values is provided to make the insertion use only one insertion strategy
     check_permutation(values_to_insert, sorted_values, sorted_values_indexes);
 
+    // TODO: Shouldn't we check `sorted_values` is sorted?
```

Comment:
@LeilaWang do you know if this is a missing check or it's indeed not needed?

Reply:
It doesn't necessary have to be sorted in increasing order.  But the values need to be arranged in a way that the low leaf will be inserted before a value if both are pending. It's on the prover to decide how to create the array, as long as it's a permutation of `values_to_insert`, which is checked above.